chrome/content/zotero/xpcom/itemTreeView.js

```diff
@@ -260,6 +261,44 @@ Zotero.ItemTreeView.prototype.setTree = Zotero.Promise.coroutine(function* (tree
 });
 
 
+Zotero.ItemTreeView.prototype.setSortColumn = function() {
+	var colId, col, dir, currentCol, currentDir;
+	
+	for (let i=0, len=this._treebox.columns.count; i<len; i++) {
+		let column = this._treebox.columns.getColumnAt(i);
+		if (column.element.getAttribute('sortActive')) {
+			currentCol = column;
+			currentDir = column.element.getAttribute('sortDirection');
+		}
+		column.element.removeAttribute('sortActive');
+		column.element.removeAttribute('sortDirection');
+	}
+	
+	colId = Zotero.Prefs.get('itemTree.sortColumnId');
+	dir = Zotero.Prefs.get('itemTree.sortDirection');
+	// Restore previous sort setting
+	if (! this.collectionTreeRow.isFeed() && colId) {
+		col = this._treebox.columns.getNamedColumn(colId);
+		dir = Zotero.Prefs.get('itemTree.sortDirection');
+		Zotero.Prefs.set('itemTree.sortColumnId', null);
+		Zotero.Prefs.set('itemTree.sortDirection', null);
```

Comment:
Overall this functions has some complicated logic, because there are 4 scenarios: 
- feed -> feed
- non-feed -> non-feed
- feed -> non-feed
- non-feed -> feed
We only want to store the sort order, if we switched from a non-feed view to a feed view. When we switch to from one feed to another, we don't want to store the previous sort-order, because it's been set to `columnAdded`, hence the check on L290. And when we restore the sort order, after switching back to a non-feed, we clear the pref, so the L290 check has `!colId == true`.